test/utils/asserts.hpp

```diff
@@ -23,45 +23,56 @@ MATCHER(Result, "empty result")
     return !arg.trapped && !arg.has_value;
 }
 
-template <typename T, typename... Ts>
-constexpr bool is_any_of = std::disjunction_v<std::is_same<T, Ts>...>;
-
 MATCHER_P(Result, value, "")  // NOLINT(readability-redundant-string-init)
 {
     using namespace fizzy;
 
-    static_assert(is_any_of<value_type, uint32_t, int32_t, uint64_t, int64_t, float, double>);
-
     // Require the arg to be TypedExecutionResult.
     // This can be a static_assert, but just returning false and failing a test provides better
     // location of the error.
     using result_type = std::remove_cv_t<std::remove_reference_t<arg_type>>;
     if constexpr (!std::is_same_v<result_type, test::TypedExecutionResult>)
+    {
+        *result_listener << "TypedExecutionResult expected";
         return false;
+    }
 
     if (arg.trapped || !arg.has_value)
         return false;
```

Comment:
Perhaps display a reason here too if you do that above?

Reply:
This is not needed because "trapped" execution is printed as "trapped", etc.

Random example:
```
Value of: execute(*instance, 0, {65534})
Expected: result 1
  Actual: trapped (of type fizzy::test::TypedExecutionResult)
```